releases.json

```diff
@@ -389,6 +389,13 @@
       "1.9-1"
     ]
   },
+  "httplib": {
+    "dependency_names": [],
```

Comment:
It provides the `httplib` dependency, you should also add it in the wrap file `[provide]` section. See https://mesonbuild.com/Wrap-dependency-system-manual.html#provide_section.

Reply:
Isn't the `httplib` dependency implicitly provided? _A wrap file named `foo.wrap` implicitly provides the dependency name `foo` even when the `[provide]` section is missing._

Reply:
Ah right, good point.

Reply:
The thing is that the provide section is checked for by the older `mesonwrap review` tooling, and

> A wrap file named foo.wrap implicitly provides the dependency name foo even when the [provide] section is missing

only counts for meson.override_dependency which, however, the wrapdb generally avoided using in preference for:

> Since that method was introduced in version 0.54.0, as a transitional aid for projects that do not yet make use of it the variable name can be provided in the wrap file with entries in the format foo-1.0 = foo_dep.


This wrap uses meson.override_dependency... and it doesn't even specify a minimum meson_version. So if people click through to https://mesonbuild.com/Wrapdb-projects.html and download the linked foo.wrap file because their meson version isn't old enough to `meson wrap install` from the v2 wrapdb... then they'll get something that just errors out. And even if they use the current version of meson to install the wrap, if they support older versions of meson then people using that older version support will hit the error.

e.g. see how the upstream PR uses `if meson.version().version_compare('>=0.54.0')`, which once this wrap is updated to pull the upstream tag will require using the [provide] section anyway.

Reply:
Hm, we never defined the minimum meson we require in wraps. Personally I don't mind just depending on latest release. But it's true that it should be set in `project()`.

Reply:
Maybe you didn't.

Reply:
@eli-schwartz you're right, I forgot to define the minimum required version..
I also thought that using override_dependency was fine since this wrap is only in WrapDB v2, that isn't completely published yet

Reply:
> The thing is that the provide section is checked for by the older `mesonwrap review` tooling

I don't think it's changed by the review tool, I copied all checks AFAIK. What should it be checking?

I personally think enough time passed, we should use `meson.override_dependency()` in our wraps. Hardcoding variable names is an horror that should stop ASAP.

Reply:
And yet... people still do use things other than the very latest meson. In fact, people still use things older than just over a year ago. The stable versions of both Debian and Ubuntu don't provide anything that new.

We're already "hardcoding" the names of `dependency_names = foo` even for newer meson, and "hardcoding" variable names in meson.build in order to pass them to meson.override_dependency() -- I don't comprehend the meaning of the sentence "Hardcoding variable names is an horror that should stop ASAP".

I'm a strong believer in being lenient in permitted versions on the grounds we would like people to actually use the wrapdb, and I still don't see a single downside...

Reply:
> I don't think it's changed by the review tool, I copied all checks AFAIK. What should it be checking?

Well, like I said, one of the review tool checks is literally `upstream.wrap has [provide] section:`. It's pedantically required by the v1 tooling, and may or may not have dependency_names vs `foo = foo_dep` but it ***must*** exist.

Reply:
Oh you're right, I did not notice that `print_status()` in reviewtool.py is fatal by default. Ok, let's make that fatal in v2 too then.